src/types/value.rs

```diff
@@ -444,6 +449,17 @@ impl DataValue {
                 buf.copy_from_slice(bytes);
                 f64::from_ne_bytes(buf)
             })),
+            LogicalType::Char(_) => {
+                // https://dev.mysql.com/doc/refman/8.0/en/char.html#:~:text=If%20a%20given%20value%20is%20stored%20into%20the%20CHAR(4)%20and%20VARCHAR(4)%20columns%2C%20the%20values%20retrieved%20from%20the%20columns%20are%20not%20always%20the%20same%20because%20trailing%20spaces%20are%20removed%20from%20CHAR%20columns%20upon%20retrieval.%20The%20following%20example%20illustrates%20this%20difference%3A
+                let value = (!bytes.is_empty()).then(|| {
+                    let last_non_zero_index = match bytes.iter().rposition(|&x| x != b' ') {
+                        Some(index) => index + 1,
+                        None => 0,
+                    };
+                    String::from_utf8(bytes[0..last_non_zero_index].to_owned()).unwrap()
+                });
```

Comment:
maybe use `trim_end()`
```rust
String::from_utf8(bytes.to_owned()).unwrap().trim_end();
```
and the link is too long 😂

Reply:
I tried it and it will make '  foo  ' become 'foo', but it should be '  foo' in mysql

Reply:
`trim_end` only remove trailing whitespace. You might used `trim()`.

Reply:
> `trim_end` only remove trailing whitespace. You might used `trim()`.

oh, you are right, but I found that `trim_end()` returns &str, which may cause String to be copied again.